tests/test_colocation_3d.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+import pickle
+
+import iris
+import numpy as np
+import pytest
+
+from pyaerocom import GriddedData
+from pyaerocom.colocation_3d import ColocatedDataLists, colocate_vertical_profile_gridded
+from tests.fixtures.data_access import TEST_DATA
+
+ROOT = TEST_DATA["Earlinet-test-3d-collocation"].path
+
+TEST_FILE = [
+    ROOT / "earlinet_example_for_ci.pkl",
+]
```

Comment:
I'm not convinced that pickeø files will work across different versions of Python and numpy/netCDF4

Reply:
I agree this is a concern. Either we find a more robust way of storing the `UngriddedData` object (not obvious to me now, would require some research), or I write a fixture to create this `UngriddedData` object on the fly. I'm happy to open a PR to patch this after the fact if you are okay merging for now